frontend/packages/dev-console/src/components/topology/TopologyDataRenderer.tsx

```diff
@@ -29,6 +40,27 @@ export const TopologyDataRenderer: React.FC<TopologyDataRendererProps> = ({
   const dataModelContext = React.useContext<ExtensibleModel>(ModelContext);
   const [model, setModel] = React.useState<Model>(null);
   const [loadError, setLoadError] = React.useState<string>(null);
+  const [monitoringAlerts, setMonitoringAlerts] = React.useState<Alerts>();
+
+  const url = PROMETHEUS_TENANCY_BASE_PATH
+    ? `${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/rules?namespace=${namespace}`
+    : null;
+  const [response, error, loading] = useURLPoll<PrometheusRulesResponse>(
+    url,
+    POLL_DELAY,
+    namespace,
+  );
+
+  const alerts: Alert[] = React.useMemo(() => {
+    if (!_.isEmpty(response)) {
+      return getAlertsAndRules(response.data).alerts;
+    }
+    return [];
+  }, [response]);
+
+  React.useEffect(() => {
+    setMonitoringAlerts({ data: alerts, loaded: !loading, loadError: error });
+  }, [alerts, loading, error]);
```

Comment:
This should not be an effect. `monitoringAlerts` is only used in one location. It can reference the values directly.

For future reference any effect that simply sets state should instead be `React.useMemo`.